queue.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+
+class QueueItem(object):
+    def __init__(self, data, prev_item=None, next_item=None):
+        self.data = data
+        self.next_item = next_item
+        self.prev_item = prev_item
+
+    def __str__(self):
+        return str(self.data)
+
+
+class Queue(object):
+    def __init__(self, first_item=None, last_item=None):
+        self.first_item = first_item
+        self.last_item = last_item
+
+    def enqueue(self, val):
+        # adds val to beginning of queue
+        new_item = QueueItem(val, next_item=self.last_item)
+        if not self.first_item:
+            self.first_item = self.last_item = new_item
+        else:
+            self.last_item.prev_item = new_item
+            self.last_item = new_item
+
+    def dequeue(self):
+        # pops last value from list and returns it
+        obsolete_item = self.first_item
```

Comment:
obsolete_item is a very confusing variable name. Why is it obsolete? Did we invent gunpowder and now our pikemen are useless? The questions are endless!